# performs ray collision checks and returns the contact points of the collisions
string sessionname
int32 sessionid

Ray[] rays

# if bodyid != 0, only checks collisions with that body
uint32 bodyid

# if 1 will also return the contacts points of each of the rays (default is 0)
byte request_contacts

---
# collision - N dim vector that is 1 for colliding rays and 0 otherwise
byte[] collision
Contact[] contacts
